adapters/httpapi_compact.c

```diff
@@ -84,6 +85,16 @@ static int ParseStringToDecimal(const char *src, int* dst)
     if (num < INT_MIN) num = INT_MIN;
     if (num > INT_MAX) num = INT_MAX;
     *dst = (int)num;
+    if (remaining && maxRemainingSize > 0)
+    {
+        size_t cLen = strlen(next);
+        if (cLen > maxRemainingSize - 1)
+        {
+            cLen = maxRemainingSize - 1;
+        }
+        (void)strcpy_s(remaining, cLen + 1, next);
+        remaining[maxRemainingSize] = 0;
```

Comment:
You can take a look at sprintf_s, and see if there is '\0' at the end if actual payload exceeded the given buffer

Reply:
sprintf_s gives a empty char[] and returns -1 if the bufSize is not enough.

I found some samples of strcpy_s added a tail `\0`